File: SustainableCommunity/src/main/java/Progmatic/SustainableCommunity/models/Item.java
package Progmatic.SustainableCommunity.models;

import Progmatic.SustainableCommunity.DTOs.ItemDTO;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import javax.persistence.*;
import java.time.LocalDateTime;


@Getter
@Setter
@Entity
public class Item {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long itemId;
    //@Column(nullable = false)
    private String itemName;
    @Enumerated(EnumType.STRING)
    private ItemCategory itemCategory;
    @Enumerated(EnumType.STRING)
    private ItemCondition itemCondition;
    private Integer itemValue;
    private Integer borrowPrice;
    @Column(length = 1200)
    private String description;
    //@Column(nullable = false)
    /**
     * Height, Width: cm-ben adja meg a User.
     */
    private Double itemHeight;
    //@Column(nullable = false)
    private Double itemWidth;
    @Lob
    private byte[] itemImage;
    private Boolean isAvailable;
    private Boolean isApproved;
    @Enumerated(EnumType.STRING)
    private ItemStatus itemStatus;
    @CreationTimestamp
    private LocalDateTime upload;
    @ManyToOne(fetch = FetchType.LAZY)
    private AppUser owner;
    @ManyToOne(fetch = FetchType.LAZY)
    private AppUser charterer;
    private Integer rateCounter = 0;
    private Double ratings;
    private Double itemRating;
    private Double wasteSavedFromLandfill;

    public Item() {
    }
/*
    public Item(String itemName, ItemCategory itemCategory, ItemCondition itemCondition, Integer itemValue,
                Integer borrowPrice, String description, Double itemHeight,
                Double itemWidth, Boolean isAvailable, Boolean isApproved, LocalDateTime upload, AppUser owner) {
        this();
        this.itemName = itemName;
        this.itemCategory = itemCategory;
        this.itemCondition = itemCondition;
        this.itemValue = itemValue;
        this.borrowPrice = borrowPrice;
        this.description = description;
        this.itemHeight = itemHeight;
        this.itemWidth = itemWidth;
        this.isAvailable = isAvailable;
        this.isApproved = isApproved;
        this.upload = upload;
        this.owner = owner;
    }
    */


    public Item(String itemName,
                ItemCategory itemCategory,
                ItemCondition itemCondition,
                Integer itemValue,
                Integer borrowPrice) {
        this.itemName = itemName;
        this.itemCategory = itemCategory;
        this.itemCondition = itemCondition;
        this.itemValue = itemValue;
        this.borrowPrice = borrowPrice;
    }

    public Item(ItemDTO itemDTO){
        this(itemDTO.getItemName(),
                itemDTO.getItemCategory(),
                itemDTO.getItemCondition(),
                itemDTO.getItemValue(),
                itemDTO.getBorrowPrice(),
                itemDTO.getDescription(),
                itemDTO.getItemHeight(),
                itemDTO.getItemWidth(),
                itemDTO.getItemImage());
    }

    public Item(String itemName,
                ItemCategory itemCategory,
                ItemCondition itemCondition,
                Integer itemValue,
                Integer borrowPrice,
                String description,
                Double itemHeight,
                Double itemWidth,
                byte[] itemImage) {
        this.itemName = itemName;
        this.itemCategory = itemCategory;
        this.itemCondition = itemCondition;
        this.itemValue = itemValue;
        this.borrowPrice = borrowPrice;
        this.description = description;
        this.itemHeight = itemHeight;
        this.itemWidth = itemWidth;
        this.itemImage = itemImage;

    }
}

File: WebSustainableCommunity/src/app/models/Item.ts
import { User } from "./User";


export enum Category {
    _name = <any>"Category",

    ADVENTURING = 0,
    CLEANING,
    COOKING_HOSTING,
    DIY,
    HOBBY,
    GARDENING,

    _labels = <any>[
        { value: Category.ADVENTURING, label: $localize `Adventuring` },
        { value: Category.CLEANING, label: "Cleaning" },
        { value: Category.COOKING_HOSTING, label: "Cooking"},
        { value: Category.DIY, label: "Diy"},
        { value: Category.HOBBY, label: "Hobby"},
        { value: Category.GARDENING, label: "Gardening"},

    ]
}


export enum Condition {
    _name = <any>"Condition",

    NEW = 0,
    VERY_GOOD,
    GOOD,
    FAIR,
    POOR,

    _labels = <any>[
        { value: Condition.NEW, label: "New" },
        { value: Condition.VERY_GOOD, label: "Very good" },
        { value: Condition.GOOD, label: "Good"},
        { value: Condition.FAIR, label: "Fair"},
        { value: Condition.POOR, label: "Poor"},

    ]
}


export class Item {
    itemId:number;
    itemName:string = "";
    itemCategory: Category;
    itemCondition: Condition;
    itemValue:number;
    borrowPrice:number;
    description:string = "";
    itemHeight:number = null;
    itemWidth:number = null;
    isAvailable:boolean;
    isApproved:boolean;
    owner:User;

}
